Special characters (also nodes containing `/`)

mbtools opened the pull request:
I added tests for handling of special characters - like slashes, quotes, and unicode - in element names, paths, and values. Everything - except of one case - was working but now we know for sure 😃 

The special case is a slash in an element name that has sub-objects: `"contains/slash": "test"` is ok, but `"contains/slash": { "test": 1 }` is not. As suggested in #169, I replaced the slash with a `tab`. Parsing now works well 👍 

For methods that require a path as input, we could either leave it as is and remember to use a tab there, too, or we pre-process the input. 

```abap
* { "contains/slash": { "test": 1 } } or better { "contains\/slash": { "test": 1 } }

r->get( '/contains/slash/test' ). " never worked

r->get( |/contains\tslash/test| ). " works with this PR

r->get( '/contains\/slash/test' ). " could work if the parameter is preprocessed i.e. replace \/ with \t
```

I would go for the middle case and not worry about the parameters. It will be a rare case anyway. 

Closes #169.

mbtools commented:
I'm not sure how you want to deal with the lint issue. It's the nature of the test to include Unicode characters. Add it as an exception?

sbcgua commented:
Yeah, old hanging issue ... I was thinking about specifiying the separator char (optionally) but it looked complex.

But what about serialization?

And did you measure the performance impact, by the way? (there should be a benchmark in the package)

linter fix - :+1:

addressing: middle case could do as a start, but would be interesting to measure perf for the 3rd one

mbtools commented:
I made a mistake making changes directly to the master of my fork. I added the serialize part. I will run some tests this week. 

FYI, the use case is again package.json where you find keys like `"@org/package": "^1.0.0"`

mbtools commented:
Going for the 3rd solution would be a big change since all methods taking a path would need to be updated for minimal benefit. 

I fixed a bug when several slashes are present and now the PR works well. 

I added "total runtime" to benchmark but can't see any significant runtime differences.

sbcgua commented:
On my hardware: `21,9010570  seconds` vs `21,9381740  seconds` after
So the diff is negligible.

Merging, Thanks for a good feature ! :+1:
